src/js/components/DataTable/Header.js

```diff
@@ -39,6 +42,38 @@ const Header = ({
   const theme = useContext(ThemeContext) || defaultProps.theme;
   return (
     <StyledDataTableHeader fillProp={fill} {...rest}>
+      {headerGroups && (
+        <TableRow>
+          {groups && <td />}
+          {columns.map(({ property }) => {
+            const headerGroup = headerGroups.groups.find(({ properties }) =>
+              properties.includes(property),
+            );
+            if (headerGroup) {
+              if (headerGroup.properties.indexOf(property) === 0) {
+                const { align, properties } = headerGroup;
+                return (
+                  <TableCell
+                    key={property}
+                    {...theme.dataTable.headerGroup}
```

Comment:
Should the styles be placed after the border/background/pad settings on lines 61-63? So the users will be able to override it from the theme.